### ISSUE TITLE
Draft release notes & release process updates for v4.4.1

### ISSUE THREAD

**Issue by ctb** [sourmash-bio/sourmash#2112]
To check:
- [x] zenodo DOI should work now!
- [x] check pip install b/c of new pyproject.toml stuff

Release process updates:
- [x] remove setuptools < 60 from initial conda create
- [x] fix `make dist`

Other fixes:
- [x] LICENSE fix on pypi
- [x] add homepage on pypi

## Release notes

Based on 
```
git log --oneline v4.4.0..latest
```

Major new features:

* less stringent size accuracy parameters for ANI accuracy reporting (#2074)
* only skip dist est if containment/jaccard are 0 or 1 (#2060)
* emit fewer warnings about potential ANI estimation issues (#2061)

Minor new features:

* fix `lca summarize` to support general collections for queries (#2107)
* add compare --avg-containment (#2056)

Documentation updates:

* fix search and gather docs (#2105)
* fix `CITATION.cff` YAML and add a test for parseability and content. (#2103)

Developer updates:

* move setup.cfg into pyproject.toml (#2097)
* Fix downsample_scaled in `core` (#2108)
* add picklist tests; support for allow_empty (#2106)
* remove LazyLoadedIndex (#2104)
* Bump web-sys from 0.3.57 to 0.3.58 (#2092)
* Bump getrandom from 0.2.6 to 0.2.7 (#2090)
* Bump wasm-bindgen-test from 0.3.30 to 0.3.31 (#2093)
* Bump pypa/cibuildwheel from 2.6.1 to 2.7.0 (#2089)
* Build: nix updates (#2088)
* CI: split wheel building (#2087)
* rust version bumps (#2086)
* Update sphinx requirement from <5,>=4.4.0 to >=4.4.0,<6 (#2068)
* Bump actions/setup-python from 3 to 4 (#2080)
* Bump myst-parser from 0.17.2 to 0.18.0 (#2081)
* Bump pypa/cibuildwheel from 2.5.0 to 2.6.1 (#2079)
* remove unnecessary `object` from `class` definitions (#2077)

**Comment by ctb** [comment]
ooh this is exciting @luizirber!! you expressed interest in how the release was gonna go after #2097 so I'll keep you updated here 😆 

end of second stage, here - https://sourmash.readthedocs.io/en/latest/release.html - 
```
% make dist
python setup.py sdist
/home/ctbrown/miniconda3/envs/sourmash-rc/lib/python3.8/distutils/dist.py:274: UserWarning: Unknown distribution option: 'milksnake_tasks'
  warnings.warn(msg)
running sdist
running egg_info
creating src/UNKNOWN.egg-info
writing src/UNKNOWN.egg-info/PKG-INFO
writing dependency_links to src/UNKNOWN.egg-info/dependency_links.txt
writing top-level names to src/UNKNOWN.egg-info/top_level.txt
writing manifest file 'src/UNKNOWN.egg-info/SOURCES.txt'
adding license file 'LICENSE' (matched pattern 'LICEN[CS]E*')
reading manifest template 'MANIFEST.in'
no previously-included directories found matching '.eggs'
warning: no previously-included files matching '*.rlib' found anywhere in distribution
warning: no previously-included files matching '*.orig' found anywhere in distribution
warning: no previously-included files matching '*.pyc' found anywhere in distribution
warning: no previously-included files matching '*.so' found anywhere in distribution
warning: no previously-included files matching '*.git' found anywhere in distribution
writing manifest file 'src/UNKNOWN.egg-info/SOURCES.txt'
running check
warning: check: missing required meta-data: name, url

warning: check: missing meta-data: either (author and author_email) or (maintainer and maintainer_email) must be supplied

creating UNKNOWN-4.4.1rc1
creating UNKNOWN-4.4.1rc1/.cargo
creating UNKNOWN-4.4.1rc1/.ci
creating UNKNOWN-4.4.1rc1/.github
creating UNKNOWN-4.4.1rc1/.github/workflows
creating UNKNOWN-4.4.1rc1/benchmarks
creating UNKNOWN-4.4.1rc1/binder
...
copying utils/trim-noV.sh -> UNKNOWN-4.4.1rc1/utils
Writing UNKNOWN-4.4.1rc1/setup.cfg
creating dist
Creating tar archive
removing 'UNKNOWN-4.4.1rc1' (and everything under it)
```

ok so some work to do here...

- [x] install milksnake
- [x] figure out the UNKNOWN problem!
- [x] figure out the missing meta-data problems - name, url, author/maintainer stuff

**Comment by ctb** [comment]
found https://github.com/pypa/setuptools_scm/issues/487, tried running `pip install -e .` and before `python setup.py sdist`, didn't help.

this seems relevant, https://setuptools.pypa.io/en/latest/build_meta.html, as does https://www.reddit.com/r/learnpython/comments/qtul8g/problems_using_pyprojecttoml/, maybe. The latter runs into the UNKNOWN problem.

## resolution

Based on the latter post, I replaced `make dist` with:
```
pip install build
python -m build --sdist
```
for both calls to `make dist`. Seems to work!

@luizirber any concerns about this solution? I'll set this up as a PR to the release instructions next.

**Comment by ctb** [comment]
note:
* I downloaded the macos arm wheel from https://github.com/sourmash-bio/sourmash/releases/tag/v4.4.1rc1 and successfully pip installed it on my laptop
* I did a `pip uninstall sourmash` after that and then transfered `sourmash-4.4.1rc1.tar.gz` over from farm (where I've been running the release). `pip install sourmash-4.4.1rc1.tar.gz` then worked as well.

So I think our release process docs are all that need to be changed, and we can proceed with the release of sourmash v4.4.1. I guess I'll give it a try and see if the conda packages work!

**Comment by ctb** [comment]
wheels built, & released! https://pypi.org/project/sourmash/4.4.1/

now we wait for the conda packages to build :)

**Comment by ctb** [comment]
yay, zenodo DOI was built! https://zenodo.org/record/6813797

**Comment by luizirber** [comment]
> @luizirber any concerns about this solution? I'll set this up as a PR to the release instructions next.

Nope, `python -m build` is the way to go! (as the [supported PyPA solution](https://pypa-build.readthedocs.io/en/latest/))

**Comment by luizirber** [comment]
Also glad to see that wheel building for the extra archs was fixed automatically somehow :upside_down_face: 
#2078

**Comment by ctb** [comment]
> Also glad to see that wheel building for the extra archs was fixed automatically somehow 🙃 #2078

I was wondering - I didn't recall that being something you had fixed but here we are!
